### ISSUE TITLE
Could not install via npm on RHEL7

### ISSUE THREAD

**Issue by tkoelpin** [nodejs/full-icu-npm#39]
Dear Developers,

due to DateTime locale formations on JavaScript, I need your full-icu, because NodeJS use a smaller version of ICU. 

I tried to install it on a RHEL7-Server with NodeJS Version 12.6.0 and NPM Version 6.10.0, but it fails twice. The NPM console gave me this error message:

````plaintext
[root@xxx ~]# npm i -g full-icu
/usr/bin/node-full-icu-path -> /usr/lib/node_modules/full-icu/node-icu-data.js

> full-icu@1.3.0 postinstall /usr/lib/node_modules/full-icu
> node postinstall.js

npm install icu4c-data@64l (Node 12.6.0 and small-icu 64.2) -> icudt64l.dat
full-icu$ /usr/bin/node /usr/lib/node_modules/npm/bin/npm-cli.js install icu4c-data@64l
Unhandled rejection Error: EACCES: permission denied, open '/root/.npm/_cacache/tmp/b9d31e8c'

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>
/usr/lib/node_modules/full-icu/install-spawn.js:41
                throw(Error(cmdPath + ' ' + args.join(' ') + ' --> status ' + spawned.status));
                ^

Error: /usr/bin/node /usr/lib/node_modules/npm/bin/npm-cli.js install icu4c-data@64l --> status 1
    at npmInstallNpm (/usr/lib/node_modules/full-icu/install-spawn.js:41:9)
    at Object.<anonymous> (/usr/lib/node_modules/full-icu/postinstall.js:72:2)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:839:10)
    at internal/main/run_main_module.js:17:11
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! full-icu@1.3.0 postinstall: `node postinstall.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the full-icu@1.3.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-07-10T06_15_05_663Z-debug.log
````

And here is the file from the 2019-07-10T06_15_05_663Z-debug.log:
````plaintext
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'i', '-g', 'full-icu' ]
2 info using npm@6.10.0
3 info using node@v12.6.0
4 verbose npm-session 49f95180d6f14335
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 304 https://registry.npmjs.org/full-icu 155ms (from cache)
8 silly pacote tag manifest for full-icu@latest fetched in 178ms
9 timing stage:loadCurrentTree Completed in 205ms
10 silly install loadIdealTree
11 silly install cloneCurrentTreeToIdealTree
12 timing stage:loadIdealTree:cloneCurrentTree Completed in 0ms
13 silly install loadShrinkwrap
14 timing stage:loadIdealTree:loadShrinkwrap Completed in 3ms
15 silly install loadAllDepsIntoIdealTree
16 silly resolveWithNewModule full-icu@1.3.0 checking installable status
17 timing stage:loadIdealTree:loadAllDepsIntoIdealTree Completed in 6ms
18 timing stage:loadIdealTree Completed in 12ms
19 silly currentTree lib
20 silly idealTree lib
20 silly idealTree └── full-icu@1.3.0
21 silly install generateActionsToTake
22 timing stage:generateActionsToTake Completed in 8ms
23 silly diffTrees action count 1
24 silly diffTrees add full-icu@1.3.0
25 silly decomposeActions action count 8
26 silly decomposeActions fetch full-icu@1.3.0
27 silly decomposeActions extract full-icu@1.3.0
28 silly decomposeActions preinstall full-icu@1.3.0
29 silly decomposeActions build full-icu@1.3.0
30 silly decomposeActions install full-icu@1.3.0
31 silly decomposeActions postinstall full-icu@1.3.0
32 silly decomposeActions finalize full-icu@1.3.0
33 silly decomposeActions refresh-package-json full-icu@1.3.0
34 silly install executeActions
35 silly doSerial global-install 8
36 verbose correctMkdir /root/.npm/_locks correctMkdir not in flight; initializing
37 verbose lock using /root/.npm/_locks/staging-a072192f34a17023.lock for /usr/lib/node_modules/.staging
38 silly doParallel extract 1
39 silly extract full-icu@1.3.0
40 silly tarball trying full-icu@latest by hash: sha512-LGLpSsbkHUT0T+EKrIJltYoejYzUqg1eW+n6wm/FTte1pDiYjeKTxO0uJvrE3jgv6V9eBzMAjF6A8jH16C0+eQ==
41 silly extract full-icu@latest extracted to /usr/lib/node_modules/.staging/full-icu-c9d3947c (29ms)
42 timing action:extract Completed in 33ms
43 silly doReverseSerial unbuild 8
44 silly doSerial remove 8
45 silly doSerial move 8
46 silly doSerial finalize 8
47 silly finalize /usr/lib/node_modules/full-icu
48 timing action:finalize Completed in 6ms
49 silly doParallel refresh-package-json 1
50 silly refresh-package-json /usr/lib/node_modules/full-icu
51 timing action:refresh-package-json Completed in 24ms
52 silly doParallel preinstall 1
53 silly preinstall full-icu@1.3.0
54 info lifecycle full-icu@1.3.0~preinstall: full-icu@1.3.0
55 timing action:preinstall Completed in 2ms
56 silly doSerial build 8
57 silly build full-icu@1.3.0
58 info linkStuff full-icu@1.3.0
59 silly linkStuff full-icu@1.3.0 has /usr/lib/node_modules as its parent node_modules
60 silly linkStuff full-icu@1.3.0 is part of a global install
61 silly linkStuff full-icu@1.3.0 is installed into a global node_modules
62 silly linkStuff full-icu@1.3.0 is installed into the top-level global node_modules
63 verbose linkBins [ { 'node-full-icu-path': './node-icu-data.js' }, '/usr/bin', true ]
64 timing action:build Completed in 6ms
65 silly doSerial global-link 8
66 silly doParallel update-linked 0
67 silly doSerial install 8
68 silly install full-icu@1.3.0
69 info lifecycle full-icu@1.3.0~install: full-icu@1.3.0
70 timing action:install Completed in 1ms
71 silly doSerial postinstall 8
72 silly postinstall full-icu@1.3.0
73 info lifecycle full-icu@1.3.0~postinstall: full-icu@1.3.0
74 verbose lifecycle full-icu@1.3.0~postinstall: unsafe-perm in lifecycle false
75 verbose lifecycle full-icu@1.3.0~postinstall: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/usr/lib/node_modules/full-icu/node_modules/.bin:/usr/lib/node_modules/.bin:/usr/local/git/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/root/bin
76 verbose lifecycle full-icu@1.3.0~postinstall: CWD: /usr/lib/node_modules/full-icu
77 silly lifecycle full-icu@1.3.0~postinstall: Args: [ '-c', 'node postinstall.js' ]
78 silly lifecycle full-icu@1.3.0~postinstall: Returned: code: 1  signal: null
79 info lifecycle full-icu@1.3.0~postinstall: Failed to exec postinstall script
80 timing action:postinstall Completed in 841ms
81 verbose unlock done using /root/.npm/_locks/staging-a072192f34a17023.lock for /usr/lib/node_modules/.staging
82 timing stage:rollbackFailedOptional Completed in 4ms
83 timing stage:runTopLevelLifecycles Completed in 1154ms
84 verbose stack Error: full-icu@1.3.0 postinstall: `node postinstall.js`
84 verbose stack Exit status 1
84 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
84 verbose stack     at EventEmitter.emit (events.js:203:13)
84 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
84 verbose stack     at ChildProcess.emit (events.js:203:13)
84 verbose stack     at maybeClose (internal/child_process.js:1021:16)
84 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
85 verbose pkgid full-icu@1.3.0
86 verbose cwd /root
87 verbose Linux 3.10.0-957.21.3.el7.x86_64
88 verbose argv "/usr/bin/node" "/usr/bin/npm" "i" "-g" "full-icu"
89 verbose node v12.6.0
90 verbose npm  v6.10.0
91 error code ELIFECYCLE
92 error errno 1
93 error full-icu@1.3.0 postinstall: `node postinstall.js`
93 error Exit status 1
94 error Failed at the full-icu@1.3.0 postinstall script.
94 error This is probably not a problem with npm. There is likely additional logging output above.
95 verbose exit [ 1, true ]
````

Oh, by the way, it belongs to the full-icu Version 1.3.0.

Thank you so much in advance for your effort.

**Comment by srl295** [comment]
It probably needs to run with elevated privilege.

**Comment by srl295** [comment]
I'm guessing that you are hitting a selinux issue with writing to `~root/.npm`

**Comment by karpikpl** [comment]
hey @tkoelpin - have you solved this issue?
I'm running install as root and getting the same error on node 14

**Comment by srl295** [comment]
probably fixed by #53

**Comment by srl295** [comment]
i think it's a dup of #17
